### PR TITLE
In CupertinoIcons doc adopt diagram and fix broken glyph map link

### PR DESCRIPTION
**Pull request by jmagman**
`CupertinoIcons` documentation currently contains a broken image, a broken link to the missing Cupertino icon gallery, and a broken link to the old `cupertino_icons` package GitHub location.  https://github.com/flutter/flutter/issues/88754

________
Screen Shot 2021-11-04 at 5 38 45 PM
________

1. Replace the broken image with a `@tool snippet`, based on the current [`Icons` snippet](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/icons.dart).  Point it to the image generated by the new `CupertinoIcons` diagram https://github.com/flutter/assets-for-api-docs/pull/172.
2. Update 0.1.3 glyph map from broken https://raw.githubusercontent.com/flutter/cupertino_icons/master/map.png to updated https://raw.githubusercontent.com/flutter/packages/master/third_party/packages/cupertino_icons/map.png.
3. Remove the link to the missing gallery.

Dependent on https://github.com/flutter/assets-for-api-docs/pull/172 first landing.
See details in https://github.com/flutter/flutter/issues/88754.
See also `cupertino_icons` package README updates at https://github.com/flutter/packages/pull/508.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/icons.dart [hunk 1]
```diff
@@ -21,11 +21,40 @@ import 'package:flutter/widgets.dart';
...
-/// For versions 1.0.0 and above (available only on Flutter SDK versions 1.22+), see the [Cupertino Icons Gallery](https://flutter.github.io/cupertino_icons).
+/// This example shows how to create a [Row] of Cupertino [Icon]s in different colors and
+/// sizes. The first [Icon] uses a [semanticLabel] to announce in accessibility
+/// modes like VoiceOver.
 ///
-/// For versions 0.1.3 and below, see this [glyph map](https://raw.githubusercontent.com/flutter/cupertino_icons/master/map.png).
+/// ![A row of icons representing a pink heart, a green bell, and a blue umbrella](https://flutter.github.io/assets-for-api-docs/assets/cupertino/cupertino_icon.png)
+///
+/// ```dart
+/// Row(
```
Comment (jmagman) [inline packages/flutter/lib/src/cupertino/icons.dart:33]: See diagram generation code at https://github.com/flutter/assets-for-api-docs/pull/172

File: packages/flutter/lib/src/cupertino/icons.dart [hunk 2]
```diff
@@ -21,11 +21,40 @@ import 'package:flutter/widgets.dart';
...
-/// [![icon gallery preview](https://raw.githubusercontent.com/flutter/cupertino_icons/master/gallery_preview_1.0.0.png)](https://flutter.github.io/cupertino_icons)
+/// {@tool snippet}
 ///
-/// For versions 1.0.0 and above (available only on Flutter SDK versions 1.22+), see the [Cupertino Icons Gallery](https://flutter.github.io/cupertino_icons).
+/// This example shows how to create a [Row] of Cupertino [Icon]s in different colors and
+/// sizes. The first [Icon] uses a [semanticLabel] to announce in accessibility
+/// modes like VoiceOver.
 ///
-/// For versions 0.1.3 and below, see this [glyph map](https://raw.githubusercontent.com/flutter/cupertino_icons/master/map.png).
+/// ![A row of icons representing a pink heart, a green bell, and a blue umbrella](https://flutter.github.io/assets-for-api-docs/assets/cupertino/cupertino_icon.png)
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/icons.dart:30]:
```suggestion
/// ![The following code snippet would generate a row of icons consisting of a pink heart, a green bell, and a blue umbrella, each progressively bigger than the last.](https://flutter.github.io/assets-for-api-docs/assets/cupertino/cupertino_icon.png)
```

**Comment by Hixie** [comment]
test-exempt: only affects comments

File: packages/flutter/lib/src/cupertino/icons.dart (hunk 2 above)
Comment (jmagman) [reply packages/flutter/lib/src/cupertino/icons.dart]: Thanks, also updated https://github.com/flutter/flutter/blob/10d1865168eafdde4e266ef5179df53160e86463/packages/flutter/lib/src/widgets/icon.dart#L33

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by jmagman** [comment]
Will wait until https://github.com/flutter/assets-for-api-docs/pull/172 is merged.

**Comment by jmagman** [comment]
https://github.com/flutter/assets-for-api-docs/pull/172 merged.

**Pull request merged**